File: golang/internal/engine/router.go
package engine

import "github.com/gofiber/fiber/v2"

func RegisterDynamicRoutes(app *fiber.App, h *Handler) {
	api := app.Group("/api")

	api.Get("/:entity", h.List)
	api.Get("/:entity/:id", h.GetByID)
	api.Post("/:entity", h.Create)
	api.Put("/:entity/:id", h.Update)
	api.Delete("/:entity/:id", h.Delete)
}

File: backend/expressjs/src/engine/router.ts
import { Router, type Express, type RequestHandler } from "express";
import type { Handler } from "./handler.js";

export function registerDynamicRoutes(
  app: Express,
  handler: Handler,
  ...middleware: RequestHandler[]
): void {
  const api = Router();

  api.get("/:entity", handler.list);
  api.get("/:entity/:id", handler.getById);
  api.post("/:entity", handler.create);
  api.put("/:entity/:id", handler.update);
  api.delete("/:entity/:id", handler.delete);

  app.use("/api", ...middleware, api);
}
